src/main/java/com/direwolf20/buildinggadgets/common/items/gadgets/GadgetDestruction.java

```diff
@@ -259,219 +206,152 @@ public static void anchorBlocks(EntityPlayer player, ItemStack stack) {
         }
     }
 
-    public static SortedSet<BlockPos> getArea(World world, BlockPos pos, EnumFacing incomingSide, EntityPlayer player, ItemStack stack) {
-        SortedSet<BlockPos> voidPositions = new TreeSet<>(Comparator.comparingInt(Vec3i::getX).thenComparingInt(Vec3i::getY).thenComparingInt(Vec3i::getZ));
-        int depth = getToolValue(stack, NBTKeys.GADGET_VALUE_DEPTH);
-        if (depth == 0)
-            return voidPositions;
-
+    public static IPositionPlacementSequence getClearingPositions(World world, BlockPos pos, EnumFacing incomingSide, EntityPlayer player, ItemStack stack) {
+        Region boundary = getClearingRegion(pos, incomingSide, player, stack);
         BlockPos startPos = (getAnchor(stack) == null) ? pos : getAnchor(stack);
-        EnumFacing side = (getAnchorSide(stack) == null) ? incomingSide : getAnchorSide(stack);
-        List<EnumFacing> directions = assignDirections(side, player);
         IBlockState stateTarget = !Config.GADGETS.GADGET_DESTRUCTION.nonFuzzyEnabled.get() || GadgetGeneric.getFuzzy(stack) ? null : world.getBlockState(pos);
+
         if (GadgetGeneric.getConnectedArea(stack)) {
-            String[] directionNames = NBTKeys.GADGET_VALUES;
-            AxisAlignedBB area = new AxisAlignedBB(pos);
-            for (int i = 0; i < directionNames.length; i++)
-                area = area.union(new AxisAlignedBB(pos.offset(directions.get(i), getToolValue(stack, directionNames[i]) - (i == 4 ? 1 : 0))));
+            HashSet<BlockPos> voidPositions = new HashSet<>();
```

Comment:
```suggestion
            Set<BlockPos> voidPositions = new HashSet<>();
```
Seems to be the only other one...